src/Catel.MVVM/Catel.MVVM.Shared/Collections/FastObservableCollection.cs

```diff
@@ -423,39 +422,49 @@ protected void NotifyChanges()
             Action action = () =>
             {
                 // Create event args
-                NotifyCollectionChangedEventArgs eventArgs = null;
+                var eventArgsList = new List<NotifyRangedCollectionChangedEventArgs>();
                 if (_suspensionContext != null && _suspensionContext.Mode == SuspensionMode.Adding)
                 {
                     if (_suspensionContext.ChangedItems.Count != 0)
                     {
-                        eventArgs = new NotifyRangedCollectionChangedEventArgs(_suspensionContext.ChangedItems, _suspensionContext.ChangedItemIndices, SuspensionMode.Adding);
+                        eventArgsList.Add(new NotifyRangedCollectionChangedEventArgs(_suspensionContext.ChangedItems, _suspensionContext.ChangedItemIndices, SuspensionMode.Adding));
                     }
                 }
                 else if (_suspensionContext != null && _suspensionContext.Mode == SuspensionMode.Removing)
                 {
                     if (_suspensionContext.ChangedItems.Count != 0)
                     {
-                        eventArgs = new NotifyRangedCollectionChangedEventArgs(_suspensionContext.ChangedItems, _suspensionContext.ChangedItemIndices, SuspensionMode.Removing);
+                        eventArgsList.Add(new NotifyRangedCollectionChangedEventArgs(_suspensionContext.ChangedItems, _suspensionContext.ChangedItemIndices, SuspensionMode.Removing));
                     }
                 }
-                else if (_suspensionContext != null && _suspensionContext.Mode == SuspensionMode.Mixed)
+                else if (_suspensionContext != null && _suspensionContext.IsMixedMode())
                 {
                     if (_suspensionContext.ChangedItems.Count != 0)
                     {
-                        eventArgs = new NotifyRangedCollectionChangedEventArgs(_suspensionContext.ChangedItems, _suspensionContext.ChangedItemIndices, _suspensionContext.MixedActions);
+                        if (_suspensionContext.Mode == SuspensionMode.Mixed)
+                        {
+                            eventArgsList.Add(new NotifyRangedCollectionChangedEventArgs(_suspensionContext.ChangedItems, _suspensionContext.ChangedItemIndices, _suspensionContext.MixedActions));
+                        }
+                        else if (_suspensionContext.Mode == SuspensionMode.MixedBash)
+                        {
+                            eventArgsList.AddRange(_suspensionContext.CreateMixedBashEventArgsList());
+                        }
                     }
                 }
                 else
                 {
-                    eventArgs = new NotifyRangedCollectionChangedEventArgs();
+                    eventArgsList.Add(new NotifyRangedCollectionChangedEventArgs());
                 }
 
```

Comment:
@alexfdezsauco Take a look at the last commit. Is this close to what you have expected? If not, can you tell me what to change?